ea-php71 and ea-php74 not shown as blockers (Version 11)

Following on from #252, ea-php71 and ea-php74 are not being shown as blockers with version 11 of the script (ea-php71 was shown as a blocker with version 7).

CentOS Linux release 7.9.2009 (Core), cPanel 110.0.5

```
# rpm -q ea-php71 ea-php74
ea-php71-7.1.33-5.5.1.cpanel.x86_64
ea-php74-7.4.33-1.1.1.cpanel.x86_64
```

```
# /scripts/elevate-cpanel --update
* 2023-05-09 16:09:28 [INFO] Self-update of script version 11 requested.
* 2023-05-09 16:09:28 [INFO] Script is up to date.
```

```
# /scripts/elevate-cpanel --check --upgrade-to=almalinux
* 2023-05-09 16:11:58 [INFO] Successfully verified signature for cpanel (key types: release).
* 2023-05-09 16:11:59 [INFO] Checking if your system is up to date: 
* 2023-05-09 16:11:59 [INFO] Running: /usr/bin/yum clean all
* 2023-05-09 16:11:59 [INFO] 
* 2023-05-09 16:11:59 [INFO] Loaded plugins: fastestmirror, universal-hooks
* 2023-05-09 16:11:59 [INFO] Cleaning repos: EA4 EA4-experimental-c7 cpanel-addons-production-feed
* 2023-05-09 16:11:59 [INFO]               : cpanel-plugins MariaDB103 base epel extras kernelcare
* 2023-05-09 16:11:59 [INFO]               : letsencrypt-cpanel updates wp-toolkit-cpanel
* 2023-05-09 16:11:59 [INFO]               : wp-toolkit-thirdparties
* 2023-05-09 16:11:59 [INFO] Cleaning up list of fastest mirrors
* 2023-05-09 16:11:59 [INFO] Other repos take up 7.4 M of disk space (use --verbose for details)
* 2023-05-09 16:11:59 [INFO] 
* 2023-05-09 16:11:59 [INFO] Running: /usr/bin/yum check-update -q
* 2023-05-09 16:11:59 [INFO] 
* 2023-05-09 16:12:24 [INFO] 
* 2023-05-09 16:12:24 [INFO] Checking /scripts/sysup
* 2023-05-09 16:12:24 [INFO] Running: /scripts/sysup
* 2023-05-09 16:12:24 [INFO] 
* 2023-05-09 16:12:25 [INFO] checkyum version 22.3  (excludes: bind-chroot)
* 2023-05-09 16:12:26 [INFO]   All Needed Packages are already installed.
* 2023-05-09 16:12:26 [INFO] 
* 2023-05-09 16:12:28 [ERROR] 1 package(s) installed from unsupported YUM repo 'letsencrypt-cpanel' from /etc/yum.repos.d/letsencrypt.repo
* 2023-05-09 16:12:28 [INFO] Unsupported YUM repo enabled 'EA4-experimental-c$releasever' without packages installed from /etc/yum.repos.d/EA4-experimental.repo, these will be disabled before ELevation
* 2023-05-09 16:12:28 [WARN] *** Elevation Blocker detected: ***
  One or more enabled YUM repo are currently unsupported and have installed packages.
  You should disable these repositories and remove packages installed from them
  before continuing the update.
  
* 2023-05-09 16:12:41 [INFO] Checking EasyApache profile compatibility with AlmaLinux 8.
* 2023-05-09 16:12:41 [INFO] Running: /usr/local/bin/ea_current_to_profile --target-os=AlmaLinux_8 --output=/root/tmp/ttUiMbNtgH/ea_profile.json
* 2023-05-09 16:12:42 [INFO] Backed up EA4 profile to /root/tmp/ttUiMbNtgH/ea_profile.json
* 2023-05-09 16:12:44 [WARN] *** Elevation Blocker detected: ***
  A package providing python36 has been detected as installed.
  This can interfere with the elevation process.
  Please remove it before elevation:
  yum remove python3
  
* 2023-05-09 16:12:44 [WARN] Please fix the detected issues before performing the elevation process.
```

Presumably this specifically is the issue now?
```
* 2023-05-09 16:12:28 [ERROR] 1 package(s) installed from unsupported YUM repo 'letsencrypt-cpanel' from /etc/yum.repos.d/letsencrypt.repo
* 2023-05-09 16:12:28 [INFO] Unsupported YUM repo enabled 'EA4-experimental-c$releasever' without packages installed from /etc/yum.repos.d/EA4-experimental.repo, these will be disabled before ELevation
* 2023-05-09 16:12:28 [WARN] *** Elevation Blocker detected: ***
  One or more enabled YUM repo are currently unsupported and have installed packages.
  You should disable these repositories and remove packages installed from them
  before continuing the update.
```
IIRC the old `letsencrypt-cpanel` being marked as a blocker is already known (see issues like #209), but we don't appear to have any plans on addressing this save for advising users to reinstall it from the cpanel-plugins repository.

The experimental EA4 repo warning is somewhat interesting, I am genuinely curious about whether this *should* be supported or not.

Otherwise, the only other blocker was for python36, which is intentional.

--- a/changelog
+++ b/changelog
@@ -1,3 +1,6 @@
+2024-XX-XX - version 20
+- RE-93 - Do not return from EA4 blocker check prematurely
+
 2024-01-16 - version 19
 - #337 - Add blocker if cPanel is not running version 110
 - #341 - Use canonical_dump() when building JSON for repo blocker reports

--- a/elevate-cpanel
+++ b/elevate-cpanel
@@ -822,7 +822,7 @@ EOS
 
         INFO("Checking EasyApache profile compatibility with $pretty_distro_name.");
 
-        return unless $self->cpev->component('EA4')->backup;          # _backup_ea4_profile();
+        $self->cpev->component('EA4')->backup;                        # _backup_ea4_profile();
         my $stash        = cpev::read_stage_file();                   # FIXME - move it to a function
         my $dropped_pkgs = $stash->{'ea4'}->{'dropped_pkgs'} // {};
         return unless scalar keys $dropped_pkgs->%*;

--- a/lib/Elevate/Blockers/EA4.pm
+++ b/lib/Elevate/Blockers/EA4.pm
@@ -38,7 +38,7 @@ sub _blocker_ea4_profile ($self) {
 
     INFO("Checking EasyApache profile compatibility with $pretty_distro_name.");
 
-    return unless $self->cpev->component('EA4')->backup;          # _backup_ea4_profile();
+    $self->cpev->component('EA4')->backup;                        # _backup_ea4_profile();
     my $stash        = cpev::read_stage_file();                   # FIXME - move it to a function
     my $dropped_pkgs = $stash->{'ea4'}->{'dropped_pkgs'} // {};
     return unless scalar keys $dropped_pkgs->%*;

--- a/t/blocker-ea4.t
+++ b/t/blocker-ea4.t
@@ -31,7 +31,7 @@ my $mock_cpev          = Test::MockModule->new('cpev');
     my $mock_isea4 = Test::MockFile->file( '/etc/cpanel/ea4/is_ea4' => 1 );
     my $type       = '';
 
-    $mock_compoment_ea4->redefine( backup => 1 );
+    $mock_compoment_ea4->redefine( backup => sub { return; } );
     my $mock_cpev = Test::MockModule->new('cpev');
     $mock_cpev->redefine(
         _read_stage_file => sub {
@@ -65,7 +65,7 @@ my $mock_cpev          = Test::MockModule->new('cpev');
 }
 
 {
-    $mock_compoment_ea4->redefine( backup => 1 );
+    $mock_compoment_ea4->redefine( backup => sub { return; } );
 
     my $ea_info_check = sub {
         message_seen( 'INFO' => "Checking EasyApache profile compatibility with AlmaLinux 8." );

